firebase-perf/src/main/java/com/google/firebase/perf/config/RemoteConfigManager.java

```diff
@@ -292,7 +306,7 @@ private void triggerRemoteConfigFetchIfNecessary() {
     if (allRcConfigMap.isEmpty()) { // Initial fetch.
       syncConfigValues(firebaseRemoteConfig.getAll());
     }
-    if (shouldFetchAndActivateRemoteConfigValues()) {
+    if (passesRandomAppStartDelay() && shouldFetchAndActivateRemoteConfigValues()) {
```

Comment:
`triggerRemoteConfigFetchIfNecessary()` gets called every time a `getRemoteConfigValue(key)` gets called, which is every time we query something like `getFloat(key)`. From my understanding from reading the code, remote config is not fetched until `FirebaseRemoteConfig.fetch()` is called. Even the `firebaseRemoteConfig.getAll()` 1 line above doesn't fetch, it should just populate `allRcConfigMap` with default values. However I am not entirely sure on that because the comment does say "Initial fetch" after all (could you or Visu confirm my understanding?). So it should be ok to be called within 5 seconds, because there are default values. Whenever another query (like `getFloat(key)`) for an RC value happens after the app start delay, the entire `allRcConfigMap` will be updated.

Reply:
That is correct. Config fetch gets initiated only when a config lookup happens. Until then, all we do is to sync with the cache.

Reply:
I'm generally not too happy with the method name here. `passRandomAppStartDelay()`. The naming convention of method should be more towards making it read it like a real sentence in the context of the caller. For eg: `hasAppStartConfigFetchDelayElapsed()`.

There are a couple of suggestions I have.
1. Rename the method like I had mentioned before `hasAppStartConfigFetchDelayElapsed()`.
2. Consider making this method called inside `shouldFetchAndActivateRemoteConfigValues()`. The `shouldFetchAndActivate...` method already takes care of validating if a fetch can be made. This initial delay is also a part of the same process. So, consider putting them together.

I would prefer option 2 if that is easier to make.

Reply:
done using option 2